R/pkg/R/generics.R

```diff
@@ -439,6 +439,10 @@ setGeneric("covar_samp", function(col1, col2) {standardGeneric("covar_samp") })
 #' @export
 setGeneric("covar_pop", function(col1, col2) {standardGeneric("covar_pop") })
 
+#' @rdname dapply
+#' @export
+setGeneric("dapply", function(x, func, schema = NULL) { standardGeneric("dapply") })
```

Comment:
OK. I will support dapply with schema only in this PR.

@shivaram, for your point 3, auto-schema detection does not work for dapply. As this happens in action (task execution phase), but schema should be aware at logical plan analyzing time.